src/Microsoft.AspNetCore.Mvc.Core/Internal/ViewEnginePath.cs

```diff
@@ -70,6 +71,11 @@ public static string ResolvePath(string path)
                     }
                     pathSegments.RemoveAt(pathSegments.Count - 1);
                 }
+                else if (segment.Equals(CurrentDirectoryToken, StringComparison.Ordinal))
```

Comment:
Can you add a test in https://github.com/aspnet/Mvc/blob/dev/test/Microsoft.AspNetCore.Mvc.Razor.Test/RazorViewEngineTest.cs#L1475 if we're changing `ViewEnginePath`